lib/extensions/ar_virtual.rb

```diff
@@ -400,12 +400,15 @@ def virtual_attribute_names
       end
     end
 
-    def attributes_builder
-      @attributes_builder ||= ::ActiveRecord::AttributeSet::Builder.new(attribute_types, primary_key) do |name|
-        unless columns_hash.key?(name) || virtual_attribute?(name)
-          _default_attributes[name].dup
```

Comment:
I don't understand what this block was doing before.  Before, we created a `Builder` object with the `attribute_types` and `primary_key`, did some dup'ing of attributes in the block.  Now, we're filtering out the `primary_key` and the `virtual attributes` and creating the `Builder` with all attributes minus these two categories of attributes.

So, why are we filtering `primary_key` when we didn't before?  Why are we now passing other non-virtual attributes to the `Builder` when we didn't before?

Reply:
What was here previously was exactly what was in the Rails codebase, with the addition of `|| virtual_attribute?(name)` (for our purposes).

I think what @kbrock did here was just do the same for the new changes that are now in `5.0.7`, which has a bug fix that we need.  Maybe some links in the description would help clear some of the confusion?

Reply:
Sure.  I can't follow what was happening before and why we changed how we instantiated the `Builder` objects without looking at the rails internals.

Reply:
The link that @d-m-u was passing around yesterday might help provide some clarity/context:

https://gitter.im/ManageIQ/manageiq/core/archives/2017/01/05

Though I personally prefer the `?at=[TIMESTAMP]` link since there is more discussion the next day that the previous link doesn't show:

https://gitter.im/ManageIQ/manageiq/core?at=586e8f5b61e516c157900687

Reply:
@jrafanie the old code is based upon rails 5.0.0 - 5.0.6
But the method changed.

I updated it to match rails 5.0.7 implementation of this method.